libs/auth/src/certs.ts

```diff
@@ -370,20 +393,39 @@ export function constructCardCertSubjectWithoutJurisdictionAndCardType(): string
   return certSubject;
 }
 
+/**
+ * Parses the provided cert and returns machine details. Throws an error if the cert doesn't follow
+ * VotingWorks's machine cert format.
+ */
+export async function parseMachineDetailsFromCert(
+  cert: Buffer
+): Promise<MachineDetails> {
+  const certDetails = await parseCert(cert);
+  assert(certDetails.component !== 'card', 'Cert is not a machine cert');
+  const { component, machineId } = certDetails;
+  return {
+    machineType: component,
+    machineId,
+    jurisdiction: component === 'admin' ? certDetails.jurisdiction : undefined,
```

Comment:
should we be stricter and fail if there's a jurisdiction and it's not a `vxadmin` ?

Reply:
This should already be enforced when we parse with our Zod schema, which indicates that VxAdmin has a jurisdiction field while other machines don't. Though I may need to switch the schemas to use a `strictObject` instead of an `object` to fully guarantee this, as a `strictObject` disallows extra fields. Will confirm

Regardless, no assertion is necessary at this stage as `certDetails.jurisdiction` is only defined if `component === 'admin'` or `component === 'card'`

Reply:
https://github.com/votingworks/vxsuite/pull/5557/commits/aeadd1da5d6c1a432fed683f73648da4bfa5e7a4